cmd/allocator/main.go

```diff
@@ -173,26 +190,43 @@ func newServiceHandler(kubeClient kubernetes.Interface, agonesClient versioned.I
 	}
 	h.caCertPool = caCertPool
 
+	tlsCert, err := readTLSCert()
+	if err != nil {
+		logger.WithError(err).Fatal("could not load TLS certs.")
+	}
+	h.tlsCert = tlsCert
```

Comment:
Since this line changes the value protected by the mutex, should the mutex be held here as well?

Reply:
Now, newServiceHandler is called to create the serviceHandler, so no one can access the object yet.